src/components/App/index.js

```diff
@@ -144,11 +132,31 @@ class App extends Component {
   receiveMessage = data => {
```

Comment:
This is turning into a big ish function - maybe we can add some comments to it :)

Reply:
Will add comments :P